src/mimeparser.rs

```diff
@@ -1398,6 +1398,17 @@ impl MimeMessage {
             } else {
                 Viewtype::File
             }
+        } else if msg_type == Viewtype::Image || msg_type == Viewtype::Gif {
+            match get_filemeta(decoded_data) {
+                // image is not too big, display as image/gif:
+                Ok((width, height)) if width * height <= constants::MAX_IMAGE_PIXELS => {
+                    part.param.set_int(Param::Width, width as i32);
```

Comment:
Maybe use `set_i64()` to get rid of the `as` operator?

Reply:
> Maybe use `set_i64()` to get rid of the `as` operator?
set_i64 does not help, since width and height are u32

Reply:
With `set_i64()` `into()` may be used then which is safe unlike `as`